### PR TITLE
feat: segment track IPs

### PR DESCRIPTION
**Pull request by antelman107**
resolves https://github.com/neondatabase/cloud/issues/17309
IP address is collected by Segment itself with `context: {direct: true}`, confirmed in Segment debugger.

### CODE REVIEW

File: src/env.ts [hunk 1]
```diff
@@ -41,3 +43,24 @@ export const getGithubEnvVars = (env: Dict<string>) => {
...
+  }
+
+  const results = [];
+  for (const netIPs of Object.entries(nets)) {
+    if (netIPs[1] === undefined) {
+      continue;
+    }
+
+    for (const ip of netIPs[1]) {
+      if (ip.family == 'IPv4') {
```
Comment (duskpoet) [inline src/env.ts:60]: why not log ipv6?

Comment (antelman107) [reply src/env.ts:60]: Moved PR to draft till the refinement.
The only thing I know is that it should be just 1 IP address. Maybe IPv6 matches as well.
image
https://segment.com/docs/connections/spec/common/#context

Comment (antelman107) [reply src/env.ts]: Now this is figured out:
image

**Review by Shridhad — APPROVED** [review]

File: src/analytics.ts [hunk 2]
```diff
@@ -74,6 +74,9 @@ export const analyticsMiddleware = async (args: {
       ci: isCi(),
       githubEnvVars: getGithubEnvVars(process.env),
     },
+    context: {
+      direct: true,
```
Comment (duskpoet) [inline src/analytics.ts:78]: does that mean that segment will collect addresses itself based on network request? I wonder which one is better in terms of accuracy...

**Pull request merged**